iroh-ffi/src/util.rs

```diff
@@ -0,0 +1,90 @@
+//! Utility functions and types.
+use std::{fmt, str::FromStr};
+
+use anyhow::Result;
+use iroh::Hash;
+
+#[repr(transparent)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Blake3Cid(pub Hash);
```

Comment:
doesn't this stuff already exist in the `iroh` code base? might as well export it, rather than duplicate this code